app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/DetailScreen.kt

```diff
@@ -24,14 +34,15 @@ fun DetailScreenRoot(
     navigation: Navigation,
     viewModel: DetailViewModel = hiltViewModel(),
 ) {
-
+    viewModel.setNavigation(navigation)
```

Comment:
That's a side-effect since you call a non-Compose function from a composable. It will execute every single time any value of the state changes possibly leading to unwanted behavior. You can fix this with effect handlers like `LaunchedEffect(true)`

Reply:
Here I set first(empty) title and description

Reply:
Yes and that's a side-effect if you just call this code in the composable without any effect handlers. It will execute any time this composable is recomposed possibly resetting the state.